sootup.callgraph/src/main/java/sootup/callgraph/AbstractCallGraphAlgorithm.java

```diff
@@ -461,17 +459,17 @@ public static Optional<? extends SootMethod> findConcreteMethod(
               + " to resolve the concrete method");
       return Optional.empty();
     }
-    Optional<? extends SootMethod> startMethod = startclass.getMethod(sig.getSubSignature());
+    Optional<SootMethod> startMethod =  startclass.getMethod(sig.getSubSignature()).map(method -> (SootMethod) method);
```

Comment:
is it possible to make SootClass.getMethod() return SootMethod (i.e. override getMethod() and cast it there) ?

Reply:
No, not if we use an Optional. 
In order to override it in the JavaSootClass with 'Optional<JavaSootClass> getMethod()', it needs to have Optional<? extends SootClass> as return type.